power_grid_model_c/power_grid_model/include/power_grid_model/component/transformer.hpp

```diff
@@ -37,7 +37,9 @@ class Transformer : public Branch {
           winding_to_{transformer_input.winding_to},
           clock_{transformer_input.clock},
           tap_side_{transformer_input.tap_side},
-          tap_pos_{transformer_input.tap_pos},
+          tap_pos_{transformer_input.tap_pos == na_IntS
+                       ? (transformer_input.tap_nom == na_IntS ? (IntS)0 : transformer_input.tap_nom)
```

Comment:
```suggestion
                       ? (transformer_input.tap_nom == na_IntS ? IntS{0} : transformer_input.tap_nom)
```

Reply:
Why this comment on this specific branch? It's written this way everywhere in the C++ code (see `tap_nom_` and `tap_direction_` for instance)

Reply:
it's a separate issue then 😬 . In general, logic like `(IntS)0` will create a 64-bit (or otherwise depending on the OS + hardware) and then reinterpret that as an 8-bit value. For `0`, that's not really an issue, but it may cause issues for other values.

Instead, `IntS{0}` or `IntS{}` will immediately create the value as 8-bit integer and initialize as `0`.

There are 2 ways to resolve this: either using incremental improvements by changing it whenever we touch the code, or by going over the entire code base in one go and change all occurrences. I'm fine either way but would like to resolve this sometime

Reply:
Thanks for the response. For now I updated all `(IntS)0` to `IntS{0}` in transformer and 3w transformer .hpp